examples/stable-diffusion/training/train_text_to_image_sdxl.py

```diff
@@ -0,0 +1,1695 @@
+#!/usr/bin/env python
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Fine-tuning script for Stable Diffusion models for text2image with support for LoRA.
+Adapted from the following sources:
+https://github.com/huggingface/diffusers/blob/v0.23.1/examples/text_to_image/train_text_to_image_sdxl.py
+https://github.com/huggingface/diffusers/blob/v0.23.1/examples/text_to_image/train_text_to_image_lora_sdxl.py
+"""
+
+import argparse
+import functools
+import gc
+import json
+import logging
+import math
+import os
+import random
+import shutil
+import time
+from pathlib import Path
+from typing import Dict
+
+import accelerate
+import datasets
+import diffusers
+import habana_frameworks.torch.core as htcore
+import numpy as np
+import torch
+import torch.nn.functional as F
+import torch.utils.checkpoint
+import transformers
+from accelerate.logging import get_logger
+from accelerate.utils import DistributedDataParallelKwargs, ProjectConfiguration
+from datasets import load_dataset
+from diffusers import (
+    AutoencoderKL,
+    UNet2DConditionModel,
+)
+from diffusers.loaders import LoraLoaderMixin
+from diffusers.optimization import get_scheduler
+from diffusers.training_utils import EMAModel, compute_snr
+from diffusers.utils import check_min_version, is_wandb_available
+from diffusers.utils.torch_utils import is_compiled_module
+from huggingface_hub import create_repo, upload_folder
+from packaging import version
+from peft import LoraConfig
+from peft.utils import get_peft_model_state_dict
+from torchvision import transforms
+from torchvision.transforms.functional import crop
+from tqdm.auto import tqdm
+from transformers import AutoTokenizer, PretrainedConfig
+
+from optimum.habana import GaudiConfig
+from optimum.habana.accelerate import GaudiAccelerator
+from optimum.habana.diffusers import GaudiEulerDiscreteScheduler, GaudiStableDiffusionXLPipeline
+from optimum.habana.utils import set_seed
+
+
+if is_wandb_available():
+    import wandb
+
+# Will error if the minimal version of diffusers is not installed. Remove at your own risks.
+check_min_version("0.23.1")
```

Comment:
```suggestion
check_min_version("0.26.0")
```
Could you also check the changes in this script between v0.23 and v0.26 and integrate them if they make sense?

Reply:
The changes are here: https://github.com/huggingface/diffusers/commit/c2717317f03b12535cfd02b477ace61189e10e4b
and we've looked at them to integrate the PEFT, otherwise 0.23 does NOT work. 
Just changed this line to 0.26.0 and it works fine. 
Will push the change.

Reply:
@imangohari1 can you update this? we want to close this PR.

Reply:
@libinta 
We pushed the changes. 
Feel free to review and move forward.